### PR TITLE
UCS-73 : Review policy/request wrappers

### PR DESCRIPTION
**Pull request by hexec**
A small simplification of the PolicyWrapper.
Solves a sonar warning for a too much complex function.

### CODE REVIEW

File: UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java [hunk 1]
```diff
@@ -68,44 +68,55 @@ public static PolicyWrapper build( String policy ) {
...
-        List<Object> list = policyType.getCombinerParametersOrRuleCombinerParametersOrVariableDefinition();
+        PolicyWrapper policyWrapper = new PolicyWrapper();
+        policyWrapper.setPolicyType( policyType );
+        policyWrapper.setPolicy( marshalPolicyType( policyType ) );
 
-        for( Object obj : list ) {
-            if( !( obj instanceof RuleType ) ) {
-                continue;
-            }
+        Reject.ifBlank( policyWrapper.getPolicy() );
```
Comment (tezcan10) [inline UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java]: Do you really need this? Because in line 79 you test policyType, can marshalling return null?

File: UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java [hunk 2]
```diff
@@ -284,8 +292,7 @@ private static PolicyType unmarshalPolicyType( String policy ) {
 
     private static String marshalPolicyType( PolicyType policy ) {
         try {
-            return JAXBUtility.marshalToString( PolicyType.class, policy, PolicyTags.POLICY,
-                JAXBUtility.SCHEMA );
+            return JAXBUtility.marshalToString( PolicyType.class, policy, PolicyTags.POLICY, JAXBUtility.SCHEMA );
         } catch( JAXBException e ) {
             log.severe( String.format( MSG_ERR_MARSHAL, e.getMessage() ) );
```
Comment (tezcan10) [inline UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java]: may be better to throw an exception?

File: UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java (hunk 1 above)
Comment (hexec) [reply UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java]: Yes since marshalling can return null for now it's needed.
It's very ugly I'll change it by handling the exceptions in those builders / sort of copy constructors :P (also ugly)

File: UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java (hunk 2 above)
Comment (hexec) [reply UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java]: yes

**Comment by hexec** [comment]
I've extended a bit this work and opted to not use Optionals in the build functions of these wrappers.
Instead I'm throwing exceptions.
The rest of the code has been adjusted, tested.
There's also a minor cleanup of the pdp
